src/rust/Dockerfile

```diff
@@ -1,174 +1,121 @@
-#
-# The Grapl Rust Dockerfile
-#
-# This file is where we build and create all our Rust images. This
-# is to avoid having to build the project dependencies multiple times
-# (e.g. once for each individual binary). Instead, we build all the
-# Rust code once and then pull out all the binaries into their own
-# images. Additionally, we leverage cargo vendor and docker's cache
-# to minimize the amount of compilation required during local builds.
-#
-# See:
-#  https://benjamincongdon.me/blog/2019/12/04/Fast-Rust-Docker-Builds-with-cargo-vendor/
-#  https://stackoverflow.com/questions/54952867/cache-cargo-dependencies-in-a-docker-volume
-#  https://github.com/rust-lang/cargo/issues/2644#issuecomment-436907777
-#  https://github.com/rust-lang/cargo/issues/2644#issuecomment-460076276
-#
-# TODO: Investigate using BuildKit to simplify this Dockerfile.
+FROM rust:1-slim-buster AS base
+
+ARG PROFILE=debug
+
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        wait-for-it \
+        wget \
+    && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /tmp
+
+RUN wget -q https://github.com/mozilla/sccache/releases/download/v0.2.15/sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz && \
+    tar xvzf sccache-v0.2.15-x86_64-unknown-linux-musl.tar.gz && \
+    chmod +x sccache-v0.2.15-x86_64-unknown-linux-musl/sccache && \
+    cp sccache-v0.2.15-x86_64-unknown-linux-musl/sccache /usr/local/bin/sccache
+
+ENV RUSTC_WRAPPER=/usr/local/bin/sccache
+
+WORKDIR /grapl
+
+# copy sources
+COPY . .
 
 #
-# Base image
+# build
 #
+# the `base` target isn't used by the build system, but let's 
+# make `base` and `build` in case it becomes convenient.
```

Comment:
This comment is confusing. I think mostly I'm confused by the word "target". Does that terminology come from referencing an image in the build section of a compose file? In the context of this dockerfile it doesn't really have a clear meaning. Here the `base` image *is* used in L32, so I'm not sure what "used by the build system" means here.

Reply:
"target" should probably be "stage" here, "target" is terminology slop I brought over from Make.

What I was trying to say is we don't need to make a new stage here. We could remove L31 and rename the first stage to "build".

Reply:
I agree the comment I have there isn't very clear. I'll try to reword.

Reply:
Fixed this in next commits